### DISCUSSION TITLE
Bootstrap responsive template

### DISCUSSION THREAD

**Discussion by fishmad** [eventum/eventum#144] [Ideas]
Ive been customizing a bootstrap responsive version of your default html templates, wondering if you interested in me contribution it to the repository?

**Comment by glensc** [comment]
yes, sure. make pull-request and then can see what to do with it (replace the default, or make multi template support)

**Comment by cpinfold** [comment]
Elan - can you create a branch from master so we can all try it?

I'm about to (finally) start making some changes to remove popup's.. it may
be better to start from the bootstrap templates.

Craig.

**Comment by glensc** [comment]
@cpinfold eee? i don't have any code to create branch. seems you never created any pull request? :)
github pull request is ticket asking to merge someones fork back to origin. so if there's pull request, there's always branch to clone somewhere :)

**Comment by fishmad** [comment]
I went with multi template so the default remained in tack for reference.

The only core change is in **init.php** on line 60:
`$define('APP_TPL_PATH', APP_PATH . '/templates');`
change to:
`$define('APP_TPL_STYLE', 'bootstrap3'); // current options: default or bootstrap3`
`$define('APP_TPL_PATH', APP_PATH . '/templates/' . APP_TPL_STYLE);`

**and folder structure in the /templates folder was **
`templates/*allfiles.tpl.html...`
**now just looks like this:**
`templates/default/*allfiles.tpl.html...`
`templates/bootstrap3/*allfiles.tpl.html...`

any new or custom style sheets and javascripts file now exist with the root js and css folders with there own sub folders named aecatly the same as template.

**Existing way:**
- root/js
- root/css

**New way:**
- root/js/**default**
- root/js/**boostrap3**
- root/css/**default**
- root/css/**boostrap3**

**Sample work in progress, you have lots of files I need to change.**
![image](https://cloud.githubusercontent.com/assets/12012101/13101328/b57d853c-d599-11e5-91cb-7b60792bf7a9.png)
![image](https://cloud.githubusercontent.com/assets/12012101/13101320/99d9f36a-d599-11e5-91b9-c0688c767e92.png)
![image](https://cloud.githubusercontent.com/assets/12012101/13101352/eef28510-d599-11e5-825a-c61386bbf04a.png)
![image](https://cloud.githubusercontent.com/assets/12012101/13101356/f62ca5d6-d599-11e5-93fe-30788c9b0def.png)
![image](https://cloud.githubusercontent.com/assets/12012101/13101366/11c1ae90-d59a-11e5-9157-01eb145febcf.png)
![image](https://cloud.githubusercontent.com/assets/12012101/13101404/7ce6b40e-d59a-11e5-93fd-cc46f45169d4.png)
![image](https://cloud.githubusercontent.com/assets/12012101/13101536/a384ef58-d59b-11e5-86fc-c4c165095c05.png)
![image](https://cloud.githubusercontent.com/assets/12012101/13101556/c1cab7e0-d59b-11e5-9dc7-ed45844da411.png)

**Comment by glensc** [comment]
looks good!
altho for simplicity of keeping the dev branch mergeable, i would not move existing templates yet.

i.e just use subdir for bootstrap ones and keep existing ones as is:

``` php
$define('APP_TPL_PATH', APP_PATH . '/templates/bootstrap3');
```

in the end run, there may still exist shared templates for both themes

**Comment by cpinfold** [comment]
@fishmad 

Looks great!

Submit a pull request when you can, I would like to add to your bootstrap templates and remove some of the pop-ups. Its something I've been meaning to do for a while.  

Thanks!

**Comment by fishmad** [comment]
I will do a pull request in the next few days.

**Comment by glensc** [comment]
btw, for validation could use this library: http://parsleyjs.org/ seems bs based, at least examples are

**Comment by glensc** [comment]
@fishmad maybe it it's too much effort for you to create pull request, you can create tarball of your work and we can try to integrate that ourselves?

**Comment by cpinfold** [comment]
@fishmad https://github.com/fishmad - the progress you made was excellent.

If it is easier, push it to your GitHub Eventum repository (
https://github.com/fishmad/eventum) and I'll rebase and create the pull
request.

Craig.

**Comment by fishmad** [comment]
Ok, I'll clone my fork an push up work

**Comment by lauripiisang** [comment]
@fishmad please try and push it into your fork so we can get this in

**Comment by infracritical** [comment]
So....question.  If I were to want to implement Eventum, which version do I use for the bootstrap version?  Do I use: https://github.com/eventum/eventum...or: https://github.com/fishmad/eventum?  Additionally which version of bootstrap was used (more curious than anything else)?

**Comment by infracritical** [comment]
One more question...  Does the bootstrap version take into consideration of custom fields?  I have the need for a tracking repository, but the focus is more so on an archived (or passive) events/incidents, rather than active tracking.  Any help would be useful.

Additionally, if I do manage to create custom fields for this modified function of Eventum, I will post commentary as an 'open' issue.

**Comment by cpinfold** [comment]
I checked a few pages of commits on the @fishmad fork and there were never
any of the bootstrap changes pushed up.  We requested a commit so someone
could continue the work.

So at the moment you need to use eventum/eventum which doesn't have the
bootstrap changes @fishmad proposed.

Craig.

**Comment by infracritical** [comment]
After searching in several places, yes, I determined that the bootstrap variant has not been released as of yet.

However, this still does not prevent me from making serious customizations for the task at hand.  Either way, I would prefer work with an application as its baseline, and customize the heck out of it, rather than develop something from scratch.  Insofar as easy-of-use, MantisBT has been the easiest for custom field development, but lacks certain properties for their prioritization.  Other packages, such as Flyspray, RT, OTRS, et. al, have proven fruitless thus far.

So...Eventum is the current selection.  If it works out (and I am **really** hoping that it does), then I will post whatever code onto this suppository...er, um...repository.

**Comment by glensc** [comment]
no code has been published with the bootstrap work, so you have to start from scratch. i hope you will publish your work as soon as possible, so could guide or help solve interesting problems coming out from that.

**Comment by glensc** [comment]
looks like we lost on this. no files/commits received from @fishmad

**Comment by glensc** [comment]
some more ideas:
- https://tailwindcss.com/
- https://postcss.org/
